### PR TITLE
Remove all but the needed curl code

### PR DESCRIPTION
**Pull request by jkrems**
The goal of this PR is to make the C++ part readable and relevant.

### CODE REVIEW

File: test/driver.test.coffee [hunk 1]
```diff
@@ -59,7 +59,7 @@ describe 'Webdriver', ->
   describe 'at timeout url', ->
     it 'throws an error when a request times out', ->
       error = assert.throws => @driver.navigateTo timeoutUrl
-      assert.include /Request connection timed out/, error.message
+      assert.include /Request timed out/, error.message
```
Comment (jkrems) [inline test/driver.test.coffee]: This was previously *always* returning connection timed out because of how http-sync was implemented.

**Comment by khoomeister** [comment]
:shipit:

**Comment by EndangeredMassa** [comment]
I'm definitely not comfortable maintaining this chunk of c++ code, but if you are, then this seems fine.

:shipit:

**Comment by jkrems** [comment]
@EndangeredMassa Since we effectively already maintained the code via `http-sync`, I'd rather maintain this reduced set than all of it. In other news: Could you do a (patch?) release? :)

**Comment by EndangeredMassa** [comment]
I don't appear to be a collaborator on this repo anymore.

**Comment by jkrems** [comment]
Dang. Adding you. Can you add me as an owner on npm in return?

**Comment by jkrems** [comment]
@EndangeredMassa You should have regained access (thanks @mcculloughsean!).

**Comment by EndangeredMassa** [comment]
published in [1.2.0](https://github.com/groupon/webdriver-http-sync/releases/tag/v1.2.0)

**Comment by EndangeredMassa** [comment]
That worked. Thanks!

I added you as an owner to this on npm.

**Comment by jkrems** [comment]
@EndangeredMassa Sure, sorry about your loss of access. Someone clicked `[x]` out of automation I guess. :)

**Comment by EndangeredMassa** [comment]
I figured. No worries.

**Pull request merged**